modules/core/packet-server/keeper/keeper.go

```diff
@@ -19,3 +27,76 @@ func NewKeeper(cdc codec.BinaryCodec, channelKeeper types.ChannelKeeper, clientK
 		ClientKeeper:  clientKeeper,
 	}
 }
+
+func (k Keeper) SendPacket(
+	ctx sdk.Context,
+	_ *capabilitytypes.Capability,
+	sourceChannel string,
+	sourcePort string,
+	destPort string,
+	timeoutHeight clienttypes.Height,
+	timeoutTimestamp uint64,
+	version string,
+	data []byte,
+) (uint64, error) {
+	// Lookup counterparty associated with our source channel to retrieve the destination channel
+	counterparty, ok := k.ClientKeeper.GetCounterparty(ctx, sourceChannel)
+	if !ok {
+		return 0, channeltypes.ErrChannelNotFound
+	}
+	destChannel := counterparty.ClientId
+
+	// retrieve the sequence send for this channel
+	// if no packets have been sent yet, initialize the sequence to 1.
+	sequence, found := k.ChannelKeeper.GetNextSequenceSend(ctx, sourcePort, sourceChannel)
+	if !found {
+		sequence = 1
+	}
+
+	// construct packet from given fields and channel state
+	packet := channeltypes.NewPacketWithVersion(data, sequence, sourcePort, sourceChannel,
+		destPort, destChannel, timeoutHeight, timeoutTimestamp, version)
+
+	if err := packet.ValidateBasic(); err != nil {
+		return 0, errorsmod.Wrap(err, "constructed packet failed basic validation")
+	}
+
+	// check that the client of receiver chain is still active
+	status := k.ClientKeeper.GetClientStatus(ctx, sourceChannel)
+	if status != exported.Active {
+		return 0, errorsmod.Wrapf(clienttypes.ErrClientNotActive, "client state is not active: %s", status)
+	}
+
+	// retrieve latest height and timestamp of the client of receiver chain
+	latestHeight := k.ClientKeeper.GetClientLatestHeight(ctx, sourceChannel)
+	if latestHeight.IsZero() {
+		return 0, errorsmod.Wrapf(clienttypes.ErrInvalidHeight, "cannot send packet using client (%s) with zero height", sourceChannel)
+	}
+
+	latestTimestamp, err := k.ClientKeeper.GetClientTimestampAtHeight(ctx, sourceChannel, latestHeight)
+	if err != nil {
+		return 0, err
+	}
+
+	// check if packet is timed out on the receiving chain
+	timeout := channeltypes.NewTimeout(packet.GetTimeoutHeight().(clienttypes.Height), packet.GetTimeoutTimestamp())
+	if timeout.Elapsed(latestHeight, latestTimestamp) {
+		return 0, errorsmod.Wrap(timeout.ErrTimeoutElapsed(latestHeight, latestTimestamp), "invalid packet timeout")
+	}
+
+	commitment := channeltypes.CommitPacket(packet)
```

Comment:
This will commit with Eureka since PacketWithVersion always creates a Eureka packet